omod-1.8/src/test/java/org/openmrs/module/webservices/rest/web/MockModuleFactoryWrapper.java

```diff
@@ -0,0 +1,110 @@
+/**
+ * This Source Code Form is subject to the terms of the Mozilla Public License,
+ * v. 2.0. If a copy of the MPL was not distributed with this file, You can
+ * obtain one at http://mozilla.org/MPL/2.0/. OpenMRS is also distributed under
+ * the terms of the Healthcare Disclaimer located at http://openmrs.org/license.
+ *
+ * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+package org.openmrs.module.webservices.rest.web;
+
+import org.openmrs.module.Module;
+import org.openmrs.module.webservices.helper.ModuleFactoryWrapper;
+import org.springframework.web.multipart.MultipartFile;
+
+import javax.servlet.ServletContext;
+import java.io.File;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+
+public class MockModuleFactoryWrapper extends ModuleFactoryWrapper {
```

Comment:
For testing purposes: all operations on modules are executed via `ModuleFactory` static methods, `Module` class is final, and actions like WebContextRefresh are failing in testing environment. Creating wrapper for `ModuleFactory` and mocking it was the easiest way to test ModuleActionResource.